src/ui/components/OverlayCard/index.js

```diff
@@ -67,3 +60,17 @@ export default class OverlayCard extends React.Component {
     );
   }
 }
+
+export const extractId = (ownProps) => {
+  return ownProps.id;
+};
+
+const OverlayCard = compose(
+  withUIState({
```

Comment:
Unless I'm missing something, this component does not need to be wrapped in `withUIState`.

It may have been confusing because the old component was implemented in a way that doesn't make sense for a React component. It was saving a ref to its child `<Overlay>` so it could show/hide the overlay but as far as I recall, nothing ever used this feature within `OverlayCard`. The React way to do it is to pass along state changes via props which it does already through `visibleOnLoad`. I think that's all it needs to do.

Reply:
I didn't initially do this or want to do this but I was getting complaints from flow. See my comment above: https://github.com/mozilla/addons-frontend/pull/5730#issuecomment-408176529.  

I'll take another look tho. Maybe there is a better way.